File: MyApplication3/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.PersistableBundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    private TextView textNumTv;
    private Button btnIncrement;
    private int counter=0;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textNumTv=findViewById(R.id.textNumTv);
        btnIncrement=findViewById(R.id.btnIncrement);
        counter=0;
        textNumTv.setText(String.valueOf(counter));
        btnIncrement.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                counter++;
                textNumTv.setText(String.valueOf(counter));
            }
        });
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putInt("key",counter);
    }

    @Override
    protected void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
        counter=savedInstanceState.getInt("key",0);
        textNumTv.setText(String.valueOf(counter));
    }
}

File: OpenNewActivity/app/src/main/java/com/example/opennewactivity/MainActivity.kt
package com.example.opennewactivity

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val secActiBut=findViewById<Button>(R.id.secActiBut)
        secActiBut.setOnClickListener {
            val intent=Intent(this,Activity2::class.java)
            startActivity(intent)
        }
    }
}